src/ray/object_manager/plasma/store.cc

```diff
@@ -912,33 +959,15 @@ Status PlasmaStore::ProcessMessage(const std::shared_ptr<Client> &client,
   uint8_t* input = (uint8_t*)message.data();
   size_t input_size = message.size();
   ObjectID object_id;
-  PlasmaObject object = {};
 
   // Process the different types of requests.
   switch (type) {
     case fb::MessageType::PlasmaCreateRequest: {
-      NodeID owner_raylet_id;
-      std::string owner_ip_address;
-      int owner_port;
-      WorkerID owner_worker_id;
-      bool evict_if_full;
-      int64_t data_size;
-      int64_t metadata_size;
-      int device_num;
-      RAY_RETURN_NOT_OK(ReadCreateRequest(
-        input, input_size, &object_id, &owner_raylet_id, &owner_ip_address, &owner_port,
-        &owner_worker_id, &evict_if_full, &data_size, &metadata_size, &device_num));
-      PlasmaError error_code = CreateObject(object_id, owner_raylet_id, owner_ip_address,
-                                            owner_port, owner_worker_id, evict_if_full,
-                                            data_size, metadata_size, device_num, client,
-                                            &object);
-      int64_t mmap_size = 0;
-      if (error_code == PlasmaError::OK && device_num == 0) {
-        mmap_size = GetMmapSize(object.store_fd);
-      }
-      RAY_RETURN_NOT_OK(SendCreateReply(client, object_id, &object, error_code, mmap_size));
-      if (error_code == PlasmaError::OK && device_num == 0) {
-        RAY_RETURN_NOT_OK(client->SendFd(object.store_fd));
+      auto status = HandleCreateObjectRequest(client, message);
```

Comment:
Can you explain more? I think it's actually okay for other objects to get created, such as if they're smaller than the objects that couldn't fit.

Reply:
Well, if there is a flood of small objects, that can prevent enough space from ever being created for large objects (e.g., a 10GB chunk).

While it may make sense to allow those to get created in the future as an optimization, how about we simplify the queueing behavior by enforcing FIFO priority for all objects?

Reply:
Anyway, I think your suggestion about just queuing all requests will address this, right?